File: 30 Days of Code/Python/day_9.py
"""
Created on Wed Jul  26 16:25:11 2017

@author: Dmitry White
"""

# TODO: Write a factorial function that takes a positive integer, N,
#  as a parameter and prints the result of N!(N factorial).

def factorial(n):
    if n == 1:
        return 1
    else:
        return n * factorial(n-1)

print(factorial(int(input())))

File: 30 Days of Code/JavaScript/day_9.js
/*
  Created on Sat May 13 16:11 2023

  @author: Dmitry White
*/

/*
  TODO: Complete the factorial function in the editor below.
  Be sure to use recursion.
  factorial has the following paramter:
  int n: an integer
*/

const factorial = (n) => {
  if (n === 1) {
    return 1;
  }
  return n * factorial(n - 1);
};

console.log('Factorial: ', factorial(10));
